Bug Fix - CXSPA-1766 - Account Summary attachment timeout

kuhris-sap opened the pull request:
Fix for bug: https://jira.tools.sap/browse/CXSPA-1766

kuhris-sap committed:
fix account summary attachment download timeout redirect bug

kuhris-sap committed:
fix account summary attachment download timeout redirect bug

kuhris-sap committed:
Resolve build error and add unit test coverage

kuhris-sap committed:
Resolve build error and add unit test coverage

kuhris-sap committed:
Merge branch 'develop' into fix/CXSPA-1766-account-summary-attachment-timeout

WeizhengSap commented on projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts:
```diff
@@ -100,4 +101,48 @@ export class AuthInterceptor implements HttpInterceptor {
+  /**
+   * If the input is a json blob of type HttpErrorResonse: extract the json error from the blob
+   * Otherwise: rethrow the error
+   */
```
I think you can create some blob error handler inside "account summary" instead of modify the existing AuthInterceptor. Please take a look of `http-interceptors` in `/organization/administration` (e.g. OrganizationBadRequestHandler). And in "account summary", you can reuse the functions in "storefrontlib".

WeizhengSap committed:
CXSPA-1766: add blob error handler

WeizhengSap committed:
merge

WeizhengSap committed:
add blobl error handler

kuhris-sap committed:
cleanup

- move blob functions out of auth interceptor
- move blob test out of auth interceptor spec
- replace function extractJsonFromBlob with FileReaderService

giancorderoortiz committed:
Merge branch 'develop' into fix/CXSPA-1766-account-summary-attachment-timeout

giancorderoortiz committed:
Merge branch 'develop' into fix/CXSPA-1766-account-summary-attachment-timeout

giancorderoortiz committed:
Merge branch 'develop' into fix/CXSPA-1766-account-summary-attachment-timeout

giancorderoortiz committed:
force rerun

kuhris-sap committed:
Merge branch 'develop' into fix/CXSPA-1766-account-summary-attachment-timeout
